src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductReferenceResolverIT.java

```diff
@@ -107,15 +107,22 @@ void sync_withNewProductWithInvalidCategoryReferences_ShouldFailCreatingTheProdu
 
         assertThat(syncStatistics).hasValues(1, 0, 0, 1, 0);
         assertThat(errorCallBackExceptions).hasSize(1);
+        assertThat(errorCallBackMessages).hasSize(1);
         final Throwable exception = errorCallBackExceptions.get(0);
-        assertThat(exception).isExactlyInstanceOf(ReferenceResolutionException.class)
-                             .hasMessageContaining("Failed to resolve 'category' resource identifier on ProductDraft "
-                                 + "with key:'productKey1'")
-                             .hasMessageContaining(format("Reason: %s", BLANK_ID_VALUE_ON_RESOURCE_IDENTIFIER));
+        final String errorMessage = errorCallBackMessages.get(0);
+        if (exception instanceof ReferenceResolutionException) {
```

Comment:
a test assertion should not expect it with if, you need to expect it directly, please add additional tests for different cases.

Reply:
The if-else block has been removed in latest change.